### PR TITLE
[Distributed] [5/N] Fix clang-tidy warnings in torch/csrc/distributed/c10d

### PR DESCRIPTION
**Pull request by cyyever**
This PR continues to fix some clang-tidy warnings in distributed/c10d code, following https://github.com/pytorch/pytorch/pull/124032.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/csrc/distributed/c10d/ProcessGroupMPI.cpp [hunk 1]
```diff
@@ -267,8 +267,8 @@ c10::intrusive_ptr<ProcessGroupMPI> ProcessGroupMPI::createProcessGroupMPI(
 
     // If no ranks are specified, assume we're creating the root group
     if (!ranks.empty()) {
-      MPI_Group worldGroup;
-      MPI_Group ranksGroup;
+      MPI_Group worldGroup = nullptr;
+      MPI_Group ranksGroup = nullptr;
```
Comment (zhuhaozhe) [inline torch/csrc/distributed/c10d/ProcessGroupMPI.cpp:271]: Hi, @cyyever.
I got a compile error here with gcc11
```
/localdisk/haozhe/pt-upstream/pytorch/torch/csrc/distributed/c10d/ProcessGroupMPI.cpp:270:30: error: cannot convert ‘std::nullptr_t’ to ‘MPI_Group’ {aka ‘int’} in initialization
  270 |       MPI_Group worldGroup = nullptr;
```
Should it be changed to:
`MPI_Group worldGroup = 0;` or some other init value?

Comment (cyyever) [reply torch/csrc/distributed/c10d/ProcessGroupMPI.cpp:271]: @zhuhaozhe  Oh, I will fix them.

**Comment by jerryzh168** [comment]
so this is related?

```
pytorch/torch/csrc/distributed/c10d/ProcessGroupMPI.cpp: In static member function ‘static c10::intrusive_ptr<c10d::ProcessGroupMPI> c10d::ProcessGroupMPI::createProcessGroupMPI(std::vector<int>)’:
pytorch/torch/csrc/distributed/c10d/ProcessGroupMPI.cpp:270:30: error: cannot convert ‘std::nullptr_t’ to ‘MPI_Group’ {aka ‘int’} in initialization
  270 |       MPI_Group worldGroup = nullptr;
      |                              ^~~~~~~
pytorch/torch/csrc/distributed/c10d/ProcessGroupMPI.cpp:271:30: error: cannot convert ‘std::nullptr_t’ to ‘MPI_Group’ {aka ‘int’} in initialization
  271 |       MPI_Group ranksGroup = nullptr;
      |                              ^~~~~~~
```

why is this error not caught by CI?

**Comment by cyyever** [comment]
> MPI_Group

It depends on the implementation, MPI_Group is a pointer type in my server. And common CI tests may not cover the MPI case.

**Pull request closed without merging**